src/drogon_http_async_writer_impl.cpp

```diff
@@ -76,6 +76,12 @@ bool DrogonHttpAsyncWriterImpl::IsDisconnected() const {
 }
 
 void DrogonHttpAsyncWriterImpl::RegisterDisconnectionCallback(std::function<void()> callback) {
+    const auto& weakConnPtr = requestPtr->getConnectionPtr();
+    if (auto connPtr = weakConnPtr.lock()) {
```

Comment:
do we need any handling if this is false?

Reply:
this could happen when the client disconnects before we even try to install callback, we can do nothing in this case